File: csharp/log-levels/LogLevels.cs
using System;
using System.Text.RegularExpressions;

static class LogLine
{
    public static string Message(string logLine) => Regex.Replace(logLine, "\\s*\\[.*\\]:\\s*", "").Trim();

    public static string LogLevel(string logLine) => Regex.Replace(logLine, "\\s*\\[(.*)\\]:.*", "$1").Trim().ToLower();

    public static string Reformat(string logLine) => $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        return logLine.replaceAll("\\s*\\[.*\\]:\\s*", "").trim();
    }

    public static String logLevel(String logLine) {
        return logLine.replaceAll("\\s*\\[(.*)\\]:.*", "$1").trim().toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}
